src/openrct2/actions/TrackPlaceAction.hpp

```diff
@@ -584,9 +583,9 @@ DEFINE_GAME_ACTION(TrackPlaceAction, GAME_COMMAND_PLACE_TRACK, TrackPlaceActionR
             }
 
             auto tileElement = tile_element_insert(
-                { mapLoc.x / 32, mapLoc.y / 32, baseZ }, quarterTile.GetBaseQuarterOccupied());
+                TileCoordsXYZ(CoordsXYZ{ mapLoc, baseZ }), quarterTile.GetBaseQuarterOccupied());
```

Comment:
A lot of this function could be simplified by using just mapLoc as it is a CoordsXYZ that has a z value that is the same as baseZ.